### DISCUSSION TITLE
[config]  what is authorityBase and b2cscopes in the example.

### DISCUSSION THREAD

**Discussion by shivika24** [stashenergy/react-native-msal#136] [General]
what is authorityBase and b2cscopes in the example?

**Comment by josmithua** [comment]
You can see their values in [this file](https://github.com/stashenergy/react-native-msal/blob/master/example/src/msalConfig.ts). You get them from your Azure Portal.

**Reply by shivika24** [reply]
What is the B2C scopes? Is it like openId? And we don’t get anything like authority base from azure, is it the authorisation url?

**Reply by shivika24** [reply]
And when adding your values it is working fine, but when I am trying adding my values which I get from Azure AD B2C it is not working… it doesn’t even open signIn/signUp page.

**Reply by ProgrammerPawan** [reply]
@shivika24 where you able you figure out, from where we can get **authorityBase**?

**Reply by shivika24** [reply]
@ProgrammerPawan, No

**Reply by ProgrammerPawan** [reply]
Thanks for you reply, but i think I just got the answer here: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/AAD-B2C-specifics

**Reply by shivika24** [reply]
I have tried using this, but this doesn’t works

**Reply by josmithua** [reply]
The value for the `authorityBase` property in the B2CClient class is basically the policy URL without the policy part.

So if this is your policy url: 

`https://{your-tenant-name}.b2clogin.com/tfp/{your-tenant-ID}/{policyname}`

Then you would use this as `authorityBase`:

`https://{your-tenant-name}.b2clogin.com/tfp/{your-tenant-ID}`

If you've set up the authority to use a custom domain, you can do that too.

For example, if this is your policy URL:

`https://login.stash.energy/tfp/1b21c465-6fe3-48f3-ba3f-de13c007d984/B2C_1A_SignInUp`

Then use this as `authorityBase`:

`https://login.stash.energy/tfp/1b21c465-6fe3-48f3-ba3f-de13c007d984`

Hope that helps!

**Reply by josmithua** [reply]
As far as scopes, you need to set those up in the app registration in your Azure Portal, under "Expose An API". You don't need to provide `openId`, `offline`, `profile` or whatever. A B2C scope should be a URL, as shown under the scope name:

![Screen Shot 2021-12-06 at 09 59 14](https://user-images.githubusercontent.com/3375071/144859449-da05aeec-1093-4209-b1f3-477c818942ad.png)

**Reply by ProgrammerPawan** [reply]
@shivika24 @josmithua By any chance do we need to register app in **Azure Active Directory B2C** and not in **Azure Active Directory** ?

**Comment by shivika24** [comment]
Thanks for the information, 1 more thing i am getting net::ERR:UNKNOWN_URL_SCHEME after signIn, signUp.
